pkg/policy/validate.go

```diff
@@ -382,40 +383,36 @@ func Validate(policy kyvernov1.PolicyInterface, client dclient.Interface, mock b
 				logging.Error(err, fmt.Sprintf("source resource %s/%s/%s not found.", rule.Generation.Kind, rule.Generation.Clone.Namespace, rule.Generation.Clone.Name))
 				continue
 			}
-
-			updateSource := true
-			label := obj.GetLabels()
-
-			if len(label) == 0 {
-				label = make(map[string]string)
-				label["generate.kyverno.io/clone-policy-name"] = policy.GetName()
-			} else {
-				if label["generate.kyverno.io/clone-policy-name"] != "" {
-					policyNames := label["generate.kyverno.io/clone-policy-name"]
-					if !strings.Contains(policyNames, policy.GetName()) {
-						policyNames = policyNames + "," + policy.GetName()
-						label["generate.kyverno.io/clone-policy-name"] = policyNames
-					} else {
-						updateSource = false
-					}
-				} else {
-					label["generate.kyverno.io/clone-policy-name"] = policy.GetName()
-				}
+			err = UpdateSourceResource(client, rule.Generation.Kind, rule.Generation.Clone.Namespace, policy, obj)
```

Comment:
Doesn't seem to be a good abstraction to have all three `rule.Generation.Kind`, `rule.Generation.Clone.Namespace` and `policy` passed into the same function. The former two originate from `policy`. 

Can we improve this?

Reply:
yes , still doing some refactoring, will push the changes in a bit.

Reply:
done

Reply:
Did you push the change? Still seeing the same absraction.

Reply:
Using the policy name instead of obj.